src/libraries/System.Runtime/tests/System/Reflection/InvokeWithRefLikeArgs.cs

```diff
@@ -0,0 +1,160 @@
+
+            Assert.Null(args[0]); // no value should have been copied back
+        }
+
+        [Fact]
+        public static void MethodTakesRefToRefStructAsArg_ThrowsNSE()
+        {
+            // Use a Binder to trick the reflection stack into treating the returned null
+            // as meaning "use the default value of the ref struct".
+
+            Mock<Binder> mockBinder = new Mock<Binder>(MockBehavior.Strict);
+            Type myRefStructType = typeof(MyRefStruct);
+            mockBinder.Setup(o => o.ChangeType("hello", myRefStructType.MakeByRefType(), null)).Returns((object)null);
+
+            MethodInfo mi = GetMethod(nameof(TestClass.TakesRefToRefStructAsArg));
+            Assert.Throws<NotSupportedException>(() => mi.Invoke(null, BindingFlags.InvokeMethod, mockBinder.Object, new object[] { "hello" }, null));
+        }
+
+        [Fact]
+        public static void MethodTakesOutToRefStructAsArg_ThrowsNSE()
+        {
+            MethodInfo mi = GetMethod(nameof(TestClass.TakesOutToRefStructAsArg));
+            Assert.Throws<NotSupportedException>(() => mi.Invoke(null, new object[] { null }));
+        }
+
+        [Fact]
+        public static void PropertyTypedAsRefToRefStruct_AsMethodInfo_ThrowsNSE()
+        {
+            MethodInfo mi = GetMethod("get_" + nameof(TestClass.PropertyTypedAsRefToRefStruct));
+            Assert.Throws<NotSupportedException>(() => mi.Invoke(null, null));
+        }
+
+        [Fact]
+        public static void PropertyTypedAsRefToRefStruct_AsPropInfo_ThrowsNSE()
+        {
+            PropertyInfo pi = typeof(TestClass).GetProperty(nameof(TestClass.PropertyTypedAsRefToRefStruct));
+            Assert.NotNull(pi);
+            Assert.Throws<NotSupportedException>(() => pi.GetValue(null));
+        }
+
+        [Fact]
+        public static void PropertyIndexerWithRefStructArg_DoesNotCopyValueBack()
+        {
+            PropertyInfo pi = typeof(TestClassWithIndexerWithRefStructArg).GetProperty("Item");
+            Assert.NotNull(pi);
+
+            object obj = new TestClassWithIndexerWithRefStructArg();
+            object[] args = new object[] { null };
+
+            object retVal = pi.GetValue(obj, args);
+            Assert.Equal(42, retVal);
+            Assert.Null(args[0]); // no value should have been copied back
+
+            pi.SetValue(obj, 42, args);
+            Assert.Null(args[0]); // no value should have been copied back
+        }
+
+        private sealed class TestClass
+        {
+            private static int _backingField = 42;
+
+            public unsafe static ref MyRefStruct ReturnsRefToRefStruct()
+            {
+                fixed (int* pInt = &_backingField)
+                {
+                    return ref *(MyRefStruct*)pInt; // will return a valid ref
+                }
+            }
+
+            public static void TakesRefStructAsArg(MyRefStruct o)
+            {
+                Assert.Equal(0, o.MyInt); // should be default(T)
```

Comment:
Do you want to set the ref struct to ensure it's not copied back?

If modified, will NotSupported be thrown, or will it just be ignored?